---
int32 color_id
string message
